src/doc/man/cargo-tree.adoc

```diff
@@ -0,0 +1,193 @@
+= cargo-tree(1)
+:idprefix: cargo_tree_
+:doctype: manpage
+:actionverb: Display
+:noall: true
+
+== NAME
+
+cargo-tree - Display a tree visualization of a dependency graph
+
+== SYNOPSIS
+
+`cargo tree [_OPTIONS_]`
+
+== DESCRIPTION
+
+This command will display a tree of dependencies to the terminal. An example of a simple project that depends on the "rand" package:
+
+----
+myproject v0.1.0 (/myproject)
+└── rand v0.7.3
+    ├── getrandom v0.1.14
+    │   ├── cfg-if v0.1.10
+    │   └── libc v0.2.68
+    ├── libc v0.2.68 (*)
+    ├── rand_chacha v0.2.2
+    │   ├── ppv-lite86 v0.2.6
+    │   └── rand_core v0.5.1
+    │       └── getrandom v0.1.14 (*)
+    └── rand_core v0.5.1 (*)
+[build-dependencies]
+└── cc v1.0.50
+----
+
+Packages marked with `(*)` have been "de-duplicated". The dependencies for the
+package have already been shown elswhere in the graph, and so are not
+repeated. Use the `--no-dedupe` option to repeat the duplicates.
+
+== OPTIONS
+
+=== Tree Options
+
+*-i*::
+*--invert*::
+    Invert the tree. Typically this flag is used with the `-p` flag to show
+    the dependencies for a specific package.
+
+*--no-dedupe*::
+    Do not de-duplicate repeated dependencies. Usually, when a package has
+    already displayed its dependencies, further occurrences will not
+    re-display its dependencies, and will include a `(*)` to indicate it has
+    already been shown. This flag will cause those duplicates to be repeated.
+
+*-d*::
+*--duplicates*::
+    Show only dependencies which come in multiple versions (implies
+    `--invert`). When used with the `-p` flag, only shows duplicates within
+    the subtree of the given package.
++
+It can be beneficial for build times and executable sizes to avoid building
+that same package multiple times. This flag can help identify the offending
+packages. You can then investigate if the package that depends on the
+duplicate with the older version can be updated to the newer version so that
+only one instance is built.
+
+*--no-dev-dependencies*::
+    Do not include dev-dependencies.
+
+*--target* _TRIPLE_::
+    Filter dependencies matching the given target-triple.
+    The default is the host platform.
+
+*--no-filter-targets*::
+    Show dependencies for all target platforms. Cannot be specified with
+    `--target`.
+
+*--graph-features*::
```

Comment:
This is so darn useful it may be nice to have a short flag for this as well perhaps? I guess `-f` is already taken, but we could always be semi-random and do like `-x`

Reply:
Sure.  `-x` is a little strange.  I'm really tempted just to steal `-f`.  How awful would that be?

Reply:
I'm not overly familiar with `cargo tree` today, but given how nice I found this I'd expect `--graph-features` to be more common to justify commandeering `-f` from `--format`.

Thinking a bit though, I wonder if we could solve a bunch of birds with one stone? What if instead of `--graph-features` we did:

```
$ cargo tree --edges features
$ cargo tree -e features
```

(assuming we rename `--dep-kinds` to `--edges`). We could change it so the defaults of `--edges` is actually the empty string. We'd basically say that there are 4 kinds of edges (build/dev/normal/features) and everything but features is enabled by default. That way `--edges normal` would go away, but you'd instead do `--edges no-build,no-dev`.

Too radical?